src/jdk.jpackage/share/classes/jdk/jpackage/internal/util/PListReader.java

```diff
@@ -204,18 +201,20 @@ public PListReader queryDictValue(String keyName) {
      *                                name in the underlying "dict" element
      */
     public boolean queryBoolValue(String keyName) {
-        final var node = getNode(keyName);
-        switch (node.getNodeName()) {
-            case "true" -> {
-                return true;
-            }
-            case "false" -> {
-                return false;
-            }
-            default -> {
-                throw new NoSuchElementException();
+        return findBoolValue(keyName).orElseThrow(NoSuchElementException::new);
+    }
+
+    public Optional<Boolean> findBoolValue(String keyName) {
+        return findNode(keyName).filter(node -> {
+            switch (node.getNodeName()) {
+                case "true", "false" -> {
+                    return true;
```

Comment:
Confused. Looks like this code will return `true` even if value is `false`. You can use `return Boolean.parseBoolean(node.getNodeName())`. No need for switch in this case.